### PR TITLE
Partial sales; Inventory sorting commands

### PR DESCRIPTION
**Pull request by GarOfMeridian**
If you try to buy too many, you'll instead get only as much as you can
hold.

Sorting:
Say 'inventory sort add scimitar' to add a scimitar, for example, to "
      "your preferences. Any item will work; you may also use item types, "
      "such as weapons, armors, helmets, shields, rings, potions, scrolls, "
      "wands, rods, reagents, ammo, and food. Say 'sort inventory' to "
      "instantly sort your inventory at any time. Say 'inventory sort clear' "
      "to delete your current preferences.

### CODE REVIEW

**Comment by skittles1** [comment]
Tested this out by buying reagents from Pacal, the partial amount given to the player isn't removed from his inventory.

**Comment by skittles1** [comment]
Have added ConsEnd and SwapListElem C functions, see #962 for descriptions and usage. Not sure if you'll need SwapListElem for this but it'll probably be useful to have.

File: kod/object/active/holder/nomoveon/battler/player/user.kod [hunk 1]
```diff
@@ -4533,6 +4557,125 @@ messages:
...
+            }
+            
+            oTemplateItem = Send(SYS,@FindTemplateItemByName,#string=String);
+            
+            if oTemplateItem <> $
+            {
+               if plClassOrderPreferences = $
+               {
+                  plClassOrderPreferences =
+                       Cons(Nth(i,2),plClassOrderPreferences);
```
Comment (skittles1) [inline kod/object/active/holder/nomoveon/battler/player/user.kod]: plClassOrderPreferences = Cons(Nth(i,2),plClassOrderPreferences);

should be

plClassOrderPreferences = Cons(GetClass(oTemplateItem),plClassOrderPreferences);

Comment (M59Gar) [inline kod/object/active/holder/nomoveon/battler/player/user.kod]: Nice catch, damn copy paste oversights

**Comment by skittles1** [comment]
You'll need to change the class header on all the reagents to Mushroom Is Reagent from Mushroom Is NumberItem etc.

**Comment by M59Gar** [comment]
nice catch

**Pull request merged**